python/pyspark/broadcast.py

```diff
@@ -139,6 +139,29 @@ def __reduce__(self):
         return _from_id, (self._jbroadcast.id(),)
 
 
+class BroadcastPickleRegistry(object):
+    """ Thread-safe registry for broadcast variables that have been pickled
+    """
+
+    def __init__(self, lock):
+        self._registry = set()
+        self._lock = lock
+
+    @property
```

Comment:
sure @HyukjinKwon, it's not really necessary.  It's just there to basically say that the lock should not be changed once this class is instantiated, so to keep it "private" but allowed to be acquired.  Maybe it's overkill here because this is not a widely used class with a very specific use.  I could remove it if that makes things easier.